File: frontend/apps/admin/src/entities/identity-review/model/identity-review.ts
export type ReviewStatus = 'pending' | 'approved' | 'rejected' | 'manual_review'

export const REVIEW_STATUS_LABELS: Record<ReviewStatus, string> = {
  pending: '대기',
  approved: '승인',
  rejected: '거절',
  manual_review: '수동심사',
}

export interface IdentityReview {
  id: string
  identityVerificationId: string
  userId: string
  userName: string
  traineeId: string | null
  verifiedName: string
  verifiedPhoneMasked: string
  status: ReviewStatus
  matchedBy: string | null
  determinedGradeId: string | null
  reviewNote: string | null
  reviewedAt: string | null
  createdAt: string
}

/** 검색·대조로 매칭할 교육생이 없을 때 — 모달에서 새로 생성해 연결 */
export interface NewTraineeInput {
  name: string
  /** 평문 수신 — 서버가 암호화 저장 */
  phone?: string
  email?: string
}

export interface ReviewApproveInput {
  /** 기존 교육생 연결 — new_trainee 와 정확히 하나만 */
  trainee_id?: string
  /** 신규 교육생 생성·연결 — 이때는 determined_grade_id 필수 */
  new_trainee?: NewTraineeInput
  /** 생략 시 기존 교육생은 현재 등급 유지. 신규 생성 시 필수 */
  determined_grade_id?: string
}

export interface ReviewRejectInput {
  review_note: string
}

File: backend/api/app/domain/identity/model/identity_review.py
import uuid
from datetime import datetime
from typing import TYPE_CHECKING

from sqlalchemy import DateTime, ForeignKey, Index, String, Text, Uuid, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.database import Base

if TYPE_CHECKING:
    from app.domain.identity.model.identity_verification import IdentityVerification
    from app.domain.user.model import User


class IdentityReview(Base):
    """본인인증 이후 기존 교육생 매칭·중복 확인·등급 판별 (비동기)."""

    __tablename__ = "identity_reviews"
    __table_args__ = (
        Index("ix_idr_user_created_at", "user_id", "created_at"),
        Index("ix_idr_trainee_id", "trainee_id"),
        Index("ix_idr_status", "status"),
    )

    id: Mapped[uuid.UUID] = mapped_column(Uuid, primary_key=True, default=uuid.uuid4)
    identity_verification_id: Mapped[uuid.UUID] = mapped_column(
        Uuid, ForeignKey("identity_verifications.id"), nullable=False
    )
    user_id: Mapped[uuid.UUID] = mapped_column(
        Uuid, ForeignKey("users.id"), nullable=False
    )
    trainee_id: Mapped[uuid.UUID | None] = mapped_column(
        Uuid, ForeignKey("trainees.id", ondelete="SET NULL"), nullable=True
    )
    status: Mapped[str] = mapped_column(String(30), nullable=False, default="pending")
    matched_by: Mapped[str | None] = mapped_column(String(30))
    determined_grade_id: Mapped[uuid.UUID | None] = mapped_column(
        Uuid, ForeignKey("membership_grades.id", ondelete="SET NULL"), nullable=True
    )
    review_note: Mapped[str | None] = mapped_column(Text)
    reviewed_by: Mapped[uuid.UUID | None] = mapped_column(
        Uuid, ForeignKey("admin_users.id", ondelete="SET NULL"), nullable=True
    )
    reviewed_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )
    user: Mapped["User | None"] = relationship("User", lazy="joined")
    identity_verification: Mapped["IdentityVerification | None"] = relationship(
        "IdentityVerification", lazy="joined"
    )
